### PR TITLE
Speedup lang-painless tests

### PR DESCRIPTION
**Pull request by rmuir**
This PR follows up on #580 to fix performance issues with `lang-painless` tests.
There are ~1000 painless test methods and each one was inefficiently creating a new script engine (re-parsing whitelists and such).
On my machine the change gives ~ 2x speedup for the `lang-painless` tests.

Now the tests that need to tweak script engine settings create a class level fixture (BeforeClass/AfterClass) that is used across all the test methods in that suite. It is also a conceptual simplification: all the test methods within a file use the same settings.

RegexLimitTests was split into two suites (limit=1 and limit=2) rather than dynamically applying different settings.

C2 compiler is no longer needed for tests to be fast, instead tests runfaster with C1 only as expected, like the rest of the unit tests.

Signed-off-by: Robert Muir <rmuir@apache.org>

### CODE REVIEW

**Comment by nknize** [comment]
start gradle check

**Comment by nknize** [comment]
Thanks @rmuir!!! These PRs are super important! Every performance improvement we can get with these tests are compounding productivity!

File: modules/lang-painless/src/main/java/org/opensearch/painless/antlr/package-info.java [hunk 1]
```diff
@@ -26,7 +26,7 @@
  */
 
 /**
- * Lexer, parser, and tree {@link Walker} responsible for turning the code
+ * Lexer, parser, and tree walker responsible for turning the code
```
Comment (rmuir) [inline modules/lang-painless/src/main/java/org/opensearch/painless/antlr/package-info.java:29]: fyi: these unrelated javadocs changes are just so that the module would compile in eclipse for me (it is picky by default about such stuff). I tried to avoid touching src/java in this PR, and I know eclipse is unsupported, but I use it for refactoring such as "rename method".

**Review by adnapibar — APPROVED** [review]
LGTM! Thanks @rmuir

**Comment by nknize** [comment]
This has sat longer than it should've because of the release tagging. Merging and closing.
